sleap_io/io/nwb.py

```diff
@@ -7,28 +7,198 @@
+                continue
+            skeleton_instances_list.append(instance_to_skeleton_instance(instance))
+
+        training_frame_skeleton_instances = SkeletonInstances(
+            skeleton_instances=skeleton_instances_list
+        )
+        training_frame_video = labeled_frame.video
+        training_frame_video_index = labeled_frame.frame_idx
+        training_frame = TrainingFrame(
+            name=training_frame_name,
+            annotator=training_frame_annotator,
+            skeleton_instances=training_frame_skeleton_instances,
+            source_video=ImageSeries(
+                name=training_frame_name,
+                description=training_frame_annotator,
+                unit="NA",
+                format="external",
+                external_file=[training_frame_video.filename],
+                dimension=[
+                    training_frame_video.backend.img_shape[0],
+                    training_frame_video.backend.img_shape[1],
+                ],
+                starting_frame=[0],
+                rate=30.0,
+            ),
+            source_video_frame_index=training_frame_video_index,
+        )
+        training_frame_list.append(training_frame)
+
+    training_frames = TrainingFrames(training_frames=training_frame_list)
+    pose_training = PoseTraining(
+        training_frames=training_frames,
+        source_videos=videos_to_source_videos(labels.videos),
+    )
+    return pose_training
+
+
+def slp_skeleton_to_nwb(skeleton: SLEAPSkeleton) -> NWBSkeleton:  # type: ignore[return]
+    """Converts SLEAP skeleton to NWB skeleton.
+
+    Args:
+        skeleton: A SLEAP skeleton.
+
+    Returns:
+        An NWB skeleton.
+    """
+    nwb_edges: list[list[int, int]]
+
+    skeleton_edges = {i: node for i, node in enumerate(skeleton.nodes)}
+    nwb_edges = []
+    for i, source in skeleton_edges.items():
+        for destination in list(skeleton_edges.values())[i:]:
+            if Edge(source, destination) in skeleton.edges:
+                nwb_edges.append([i, list(skeleton_edges.values()).index(destination)])
+
+    return NWBSkeleton(
+        name=f"Nodes {skeleton.nodes[0].name}, ..., {skeleton.nodes[-1].name}",
+        nodes=skeleton.node_names,
+        edges=np.array(nwb_edges, dtype=np.uint8),
+    )
+
+
+def instance_to_skeleton_instance(instance: Instance) -> SkeletonInstance:  # type: ignore[return]
+    """Converts a SLEAP Instance to an NWB SkeletonInstance.
+
+    Args:
+        instance: A SLEAP Instance.
+
+    Returns:
+        An NWB SkeletonInstance.
+    """
+    skeleton = slp_skeleton_to_nwb(instance.skeleton)
+    points_list = list(instance.points.values())
+    node_locs = [[point.x, point.y] for point in points_list]
+    np_node_locations = np.array(node_locs)
+    return SkeletonInstance(
+        name=name_generator("skeleton_instance"),
+        id=np.uint(10),
+        node_locations=np_node_locations,
+        node_visibility=[True for _ in range(len(skeleton.nodes))],
```

Comment:
No! Visibility is defined in our `Point` objects: https://io.sleap.ai/latest/reference/sleap_io/model/instance/#sleap_io.model.instance.Point

Pull this out when iterating over `instance.points` above.